src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/MissingGeneratorAttributeRuleTests.cs

```diff
@@ -0,0 +1,352 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.SourceGeneratorAttributeAnalyzer,
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.SourceGeneratorAttributeAnalyzerFix>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.SourceGeneratorAttributeAnalyzer,
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.SourceGeneratorAttributeAnalyzerFix>;
+
+namespace Microsoft.CodeAnalysis.Analyzers.UnitTests.MetaAnalyzers
+{
+    public class MissingGeneratorAttributeRuleTests
+    {
+        private static readonly ReferenceAssemblies ReferenceAssemblies =
+            ReferenceAssemblies.Default.AddPackages(ImmutableArray.Create(new PackageIdentity("Microsoft.CodeAnalysis", "3.10.0")));
+
+        [Fact]
+        public async Task TestSimpleClass_CSharp()
+        {
+            var code = @"
+using Microsoft.CodeAnalysis;
+
+public class [|CustomGenerator|] : ISourceGenerator
+{
+    public void Initialize(GeneratorInitializationContext context) {}
+    public void Execute(GeneratorExecutionContext context) {}
+}";
+
+            var fixedCode = @"
+using Microsoft.CodeAnalysis;
+
+[Generator]
+public class CustomGenerator : ISourceGenerator
+{
+    public void Initialize(GeneratorInitializationContext context) {}
+    public void Execute(GeneratorExecutionContext context) {}
+}";
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources = { code },
+                    ReferenceAssemblies = ReferenceAssemblies
+                },
+                FixedState =
+                {
+                    Sources = { fixedCode },
+                    ReferenceAssemblies = ReferenceAssemblies
+                },
```

Comment:
💡 `FixedState` inherits unspecified items from `TestState`. Also, this form is typically used for cases where a code fix is adding/removing references. For these tests, you can set `Test.ReferenceAssemblies`:

```suggestion
                ReferenceAssemblies = ReferenceAssemblies,
                TestState =
                {
                    Sources = { code },
                },
                FixedState =
                {
                    Sources = { fixedCode },
                },
```